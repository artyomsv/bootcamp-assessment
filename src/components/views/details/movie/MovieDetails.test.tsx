import * as React from 'react';
import {createShallow} from '@material-ui/core/test-utils';
import MovieDetailsView from './MovieDetails.view';

const shallow = createShallow();

describe('renders without crashing', () => {
  test('if renders', () => {
    const mockFn = jest.fn();
    const result = shallow(
      <MovieDetailsView
        isFetching={false}
        fetchMovieDetails={mockFn}
        clearState={mockFn}
        details={{} as any}
        navigateToActor={mockFn}
        goBack={mockFn}
      />
    );

    expect(result).toMatchSnapshot();
  });
});
