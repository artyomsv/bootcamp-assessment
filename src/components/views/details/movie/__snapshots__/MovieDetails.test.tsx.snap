// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing if renders 1`] = `
<withRouter(MovieDetailsView)
  actors={Array []}
  classes={
    Object {
      "button": "withRouter-MovieDetailsView--button-7",
      "companyImage": "withRouter-MovieDetailsView--companyImage-4",
      "demo": "withRouter-MovieDetailsView--demo-9",
      "details": "withRouter-MovieDetailsView--details-2",
      "extendedIcon": "withRouter-MovieDetailsView--extendedIcon-8",
      "image": "withRouter-MovieDetailsView--image-3",
      "linkImage": "withRouter-MovieDetailsView--linkImage-5",
      "movies": "withRouter-MovieDetailsView--movies-6",
      "root": "withRouter-MovieDetailsView--root-1",
    }
  }
  clearState={[MockFunction]}
  details={Object {}}
  fetchMovieActors={[MockFunction]}
  fetchMovieDetails={[MockFunction]}
  goBack={[MockFunction]}
  isActorsFetching={false}
  isMovieFetching={false}
  navigateToActor={[MockFunction]}
  navigateToMovie={[MockFunction]}
/>
`;
